### ISSUE TITLE
The difference between the implementation of SETR and the official open source code

### ISSUE THREAD

**Issue by sixiaozheng** [open-mmlab/mmsegmentation#721]
Hello, I am the author of SETR. Thank you very much for your team's efforts to reproduce SETR in mmsegmentation. I noticed that the re-implementation performance is somewhat different than ours. After comparing the code and configuration files, I found the following differences.

**1. SETR-Naive**
In our implementation, both decode_head and auxiliary_head only perform upsampling operations once, but the implementation in mmsegmentation is that decode_head and auxiliary_head both perform upsampling operations twice.

**2. SETR-PUP**
The auxiliary_head of SETR-PUP is the same as the auxiliary_head of SETR-Naïve. In our implementation, auxiliary_head only performs one upsampling operation, but in mmsegmentation, auxiliary_head performs two upsampling operations.

**our implementation：**
https://github.com/fudan-zvg/SETR/blob/23f8fde88182c7965e91c28a0c59d9851af46858/mmseg/models/decode_heads/vit_up_head.py#L79
![image](https://user-images.githubusercontent.com/34769435/126891994-25810282-a41d-474f-a031-82c4cf975867.png)

**mmsegmentation config information：**
![image](https://user-images.githubusercontent.com/34769435/126892020-a7eb54bb-cba3-44dc-b395-004f7f79572c.png)

**Comment by Junjun2016** [comment]
Hi @sennnnn 
Please have a check and train it again.

**Comment by sennnnn** [comment]
If keep the raw architecture, the code may be hacky. So, we do some modifications.

**Comment by sennnnn** [comment]
## About the auxiliary heads

The reason why we remove the fourth auxiliary head of PUP is that the fourth auxiliary head may have similar gradient with decode head. The fourth auxiliary head may not bring improvement of performance.

## About the upsampling operations

This is the setr-naive config:

```python
# model settings
backbone_norm_cfg = dict(type='LN', eps=1e-6, requires_grad=True)
norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained=\
    'https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-vitjx/jx_vit_large_p16_384-b3be5167.pth',  # noqa
    backbone=dict(
        type='VisionTransformer',
        img_size=(768, 768),
        patch_size=16,
        in_channels=3,
        embed_dims=1024,
        num_layers=24,
        num_heads=16,
        out_indices=(9, 14, 19, 23),
        drop_rate=0.1,
        norm_cfg=backbone_norm_cfg,
        with_cls_token=True,
        interpolate_mode='bilinear',
    ),
    decode_head=dict(
        type='SETRUPHead',
        in_channels=1024,
        channels=256,
        in_index=3,
        num_classes=19,
        dropout_ratio=0,
        norm_cfg=norm_cfg,
        num_convs=1,
        up_scale=4,
        kernel_size=1,
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    auxiliary_head=[
        dict(
            type='SETRUPHead',
            in_channels=1024,
            channels=256,
            in_index=0,
            num_classes=19,
            dropout_ratio=0,
            norm_cfg=norm_cfg,
            num_convs=1,
            up_scale=4,
            kernel_size=1,
            align_corners=False,
            loss_decode=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)),
        dict(
            type='SETRUPHead',
            in_channels=1024,
            channels=256,
            in_index=1,
            num_classes=19,
            dropout_ratio=0,
            norm_cfg=norm_cfg,
            num_convs=1,
            up_scale=4,
            kernel_size=1,
            align_corners=False,
            loss_decode=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)),
        dict(
            type='SETRUPHead',
            in_channels=1024,
            channels=256,
            in_index=2,
            num_classes=19,
            dropout_ratio=0,
            norm_cfg=norm_cfg,
            num_convs=1,
            up_scale=4,
            kernel_size=1,
            align_corners=False,
            loss_decode=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4))
    ],
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
```

* The decode head of SETR-Naive only has one conv with one upsampling.
